storage/engine/rocksdb/db.cc

```diff
@@ -39,33 +39,84 @@
 extern "C" {
 #include "_cgo_export.h"
 
-struct DBBatch {
+struct DBEngine {
+  rocksdb::DB* const rep;
```

Comment:
This pointer in the base class doesn't sit quite right with me. I'd rather have each of the subclasses contain their own pointer to a `DB`. `EngineGetter` is the only thing that needs a `rep` pointer on a `DBEngine` of unknown subtype, and I think I'd rather move that part of `EngineGetter` onto a new method of `DBEngine` that can abstract away both the `rep` pointer and the snapshot passed to `MakeReadOptions`.